s3torchconnector/tst/unit/test_s3_client.py

```diff
@@ -82,3 +88,36 @@ def test_user_agent_always_starts_with_package_version(comments):
     if comments_str:
         assert comments_str in s3_client.user_agent_prefix
         assert comments_str in s3_client._client.user_agent_prefix
+
+
+@given(
+    part_size=integers(min_value=5 * MiB, max_value=5 * GiB),
+    throughput_target_gbps=floats(min_value=10.0, max_value=100.0),
+)
+@example(part_size=5 * MiB, throughput_target_gbps=10.0)
+@example(part_size=5 * GiB, throughput_target_gbps=15.0)
+def test_s3_client_custom_config(part_size: int, throughput_target_gbps: float):
+    # Part size must have values between 5MiB and 5GiB
+    s3_client = S3Client(
+        region=TEST_REGION,
+        s3client_config=S3ClientConfig(
+            part_size=part_size,
+            throughput_target_gbps=throughput_target_gbps,
+        ),
+    )
+    assert s3_client._client.part_size == part_size
+    assert s3_client._client.throughput_target_gbps == throughput_target_gbps
+
+
+@pytest.mark.parametrize("part_size", [1, 2 * KiB, 6 * GiB])
+def test_s3_client_invalid_part_size_config(part_size: int):
+    with pytest.raises(
+        S3Exception,
+        match="invalid configuration: part size must be at between 5MiB and 5GiB",
+    ):
+        s3_client = S3Client(
+            region=TEST_REGION,
+            s3client_config=S3ClientConfig(part_size=part_size),
+        )
+        # The client is lazily initialized
+        assert s3_client._client.part_size is not None
```

Comment:
Couldn't we do an == check here?

Reply:
It fails before getting to verify the assert, when is trying to initialize MP client with the invalid part_size value, that's why I avoided the == assertion.